server/app/models/template.js

```diff
@@ -101,9 +100,8 @@ class TemplateUnit extends Model {
         // eslint-disable-next-line
         set(val) {
           const date = new Date();
-          const currDay = date.getDay();
 
-          date.setDate(date.getDate() + (val.day - 1 - currDay));
+          date.setDate(date.getDate() + ((7 + val.day - 1 + date.getDay()) % 7));
```

Comment:
I change your implementation because it doesn't work with the transformation from Number to number in the front side. I dont really understand why...

This implementation set date to the next weekday in the mounth. Anyway, it works now ;)

I follow this explanation: https://codereview.stackexchange.com/a/33648